---
title: "MICB425_portfolio"
author: "Jawairia Atif, 39139134"
date: "version `r format(Sys.time(), '%B %d, %Y')`"
output:
  html_document:
    toc: yes
    toc_depth: 3
  pdf_document:
    latex_engine: xelatex
    toc: yes
    toc_depth: '3'
  word_document:
    toc: yes
    toc_depth: '3'
---

# Module 01


## Module 01 portfolio check

* Installation check
    + Completion status:
    + Comments:
* Portfolio repo setup
    + Completion status: 
    + Comments:
* RMarkdown Pretty PDF Challenge
    + Completion status:
    + Comments:
* Evidence worksheet_01
    + Completion status:
    + Comments:
* Evidence worksheet_02
    + Completion status:
    + Comments:
* Evidence worksheet_03
    + Completion status:
    + Comments:
* Problem Set_01
    + Completion status:
    + Comments:
* Problem Set_02
    + Completion status:
    + Comments:
* Writing assessment_01
    + Completion status:
    + Comments:
* Additional Readings
    + Completion status:
    + Comments  


## Data science Friday

### Installation check

![Screenshot of GIT console](c:/Users/Jawairia/Documents/MICB425_portfolio/Screenshot_GIT.png)

![Screenshot of GitHub home page](c:/Users/Jawairia/Documents/MICB425_portfolio/Screenshot_GitHub.png)

![Screenshot of Rstudio](c:/Users/Jawairia/Documents/MICB425_portfolio/Screenshot_Rstudio.png)



### Portfolio repo setup

**Creating the direcrory**
cd ~/documents   
mkdir MICB425_portfolio   
touch ID.txt

**Initializing repo on Github**
git init   git add .   
git commit -m"first commit"   
git remote add origin https://remote_repository_URL 
git remote -v   
git push -u origin master 

**Pushing to Github**

git add . 
git commit -m "comment"
git status
git push

### RMarkdown pretty PDF challenge


---
title: "Pretty_html"
author: "Jawairia Atif (39139134)"
date: "version `r format(Sys.time(), '%B %d, %Y')`"
output:
  html_document:
    toc: yes
---

# R Markdown PDF Challenge
The following assignment is an exercise for the reproduction of this .html document using the RStudio and RMarkdown tools we've shown you in class. Hopefully by the end of this, you won't feel at all the way this poor PhD student does. We're here to help, and when it comes to R, the internet is a really valuable resource. This open-source program has all kinds of tutorials online.


![http://phdcomics.com/ Comic posted 1-17-2018](c:/Users/Jawairia/Documents/MICB425_portfolio/phd011718s.gif)

## Challenge Goals

The goal of this R Markdown html challenge is to give you an opportunity to play with a bunch of different RMarkdown formatting. Consider it a chance to flex your RMarkdown muscles. Your goal is to write your own RMarkdown that rebuilds this html document as close to the original as possible. So, yes, this means you get to copy my irreverant tone exactly in your own Markdowns. It's a little window into my psyche. Enjoy =)

**hint: go to the [PhD Comics website](http://phdcomics.com/) to see if you can find the image above**  
*If you can't find that exact image, just find a comparable image from the PhD Comics website and include it in your markdown*

### Here's a header!

Let's be honest, this header is a little arbitrary. But show me that you can reproduce headers with different levels please. This is a level 3 header, for your reference (you can most easily tell this from the table of contents)

#### Another header, now with maths

Perhaps you're already really confused by the whole markdown thing. Maybe you're so confused that you've forgotton how to add. Never fear! ~~A calculator~~ R is here:

``` {r}
1231521+12341556280987
```
### Table Time

Or maybe, after you've added those numbers, you feel like it's about time for a table!  
I'm going to leave all the guts of the coding here so you can see how libraries (R packages) are loaded into R (more on that later). It's not terribly pretty, but it hints at how R works and how you will use it in the future. The summary function used below is a nice data exploration function that you may use in the ^future^.

```{r}
library(knitr)
kable(summary(cars),caption="I made this table with kable in the knitr package library")
```

And now you've almost finished your first RMarkdown! Feeling excited? We are! In fact, we're so excited that maybe we need a big finale eh? Here's ours! Include a fun gif of your choice!

![meow](c:/Users/Jawairia/Documents/MICB425_portfolio/meow.gif)  

### Data science assignment 4
```{r message=FALSE, warning=FALSE, paged.print=TRUE}
# install.packages("tidyverse")
library(tidyverse)

metadata <- read.table(file="c:/Users/Jawairia/Documents/MICB425_portfolio/Saanich.metadata.txt", header=TRUE, row.names=1, sep="\t", na.strings="NAN")
otu= read.table(file="c:/Users/Jawairia/Documents/MICB425_portfolio/saanich.otu.txt", header=TRUE, row.names=1, sep="\t", na.strings="NAN")
# source("https://bioconductor.org/biocLite.R")
# biocLite("phyloseq")

library(phyloseq)

load("phyloseq_object.RData")
```


####Exercise 1
```{r fig.height=4, fig.width=6}
#Figure 1: Plotting NO3_uM against depth
ggplot(metadata, aes(x=NO3_uM, y=Depth_m)) + 
  geom_point(shape=17, color="purple")
```

####Exercise 2
```{r fig.height=4, fig.width=6}
#Figure 2: 
metadata %>% 
  mutate(Temperature_F = Temperature_C*1.8 +32) %>% 
  ggplot() + geom_point(aes(x=Temperature_F, y=Depth_m))+
  labs(x= "Temperature (F)", y="Depth (m)")

```

####Exercise 3
```{r fig.height=9, fig.width=5}
#Figure 3:
physeq_percent = transform_sample_counts(physeq, function(x) 100 * x/sum(x)) 
  plot_bar(physeq_percent, fill="Class") + 
  geom_bar(aes(fill=Class), stat="identity") + labs(x="Sample depth", y="Percent relative abundance")+
    theme(legend.position = "bottom" ,  legend.text=element_text(size=8), plot.title = element_text(size=12, face="bold", hjust=0.5), axis.text= element_text(size=9))+
  guides(fill=guide_legend(ncol=3,bycol=TRUE))

```

####Exercise 4

```{r fig.height=6, fig.width=8, message=FALSE, warning=FALSE, paged.print=FALSE}
#Figure 4: 
library(tidyverse)
metadata %>% 
  select(Depth_m, O2_uM, PO4_uM, SiO2_uM, NO3_uM, NH4_uM, NO2_uM) %>%
  gather("Nutrients","uM", O2_uM:NO2_uM)%>%
  ggplot() + geom_point(aes(x=Depth_m, y=uM)) +
  facet_wrap(~Nutrients, scales="free_y") +
  theme(legend.position="none")

```

## Origins and Earth Systems

### Evidence worksheet 01 
Paper:
[Whitman *et al* 1998](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC33863/)

#### Learning objectives
Describe the numerical abundance of microbial life in relation to ecology and biogeochemistry of Earth systems. 

#### General questions
#### What were the main questions being asked?

The main questions were: 

* To determine the number of prokaryotes in different habitats and how much they contribute to the total number of microbes on Earth.
* What are the turnover rates of the microbes in these environments and which habitats are the most productive?
* What is the total amount of cellular carbon and other nutrients (N, P) produced by these prokaryotes on earth?
* Estimate prokaryotic diversity in different habitats.

#### What were the primary methodological approaches used?

* Pooling data from studies in which there is sampling of prokaryotes from different habitats, and quantification of cells in these samples and extrapolation, estimations, assumptions, mathematical formulas to calculate cell numbers, nutrient contents etc.:   

* Soil: 
  +estimating prokaryotic abundance in differant habitats  
    + ocean below 200m
        + average cellular density and multiplied it with the estimated amount of marine and fresh water quantities
    + different soils
        + estimations cellular density from direct cell counts from different soils in past field studies.
        + calculate number of microbes by looking at perviously estimated amounts of soil on earth
    + terrestrial subsurface in various dephts
        + estimated prokaryotic numbers from ground water based on values of several sites, and multiplied this using the estimated value of ground water on Earth. 
        - looked at the average porosity of Earth's soil and used known values of space occupied by prokaryotes in these pores
        
    + Animals and other habitats
        + human: skin surface area multiplied by cell density of prokaryotes on the skin  
		    + insects: using termites as a model: multiple number of microbes in insect by total number of insect on the planet.
		    + leaves: leaf area x assumed dense population of microbes  
		    - air: pre-calculated data used
    

#### Summarize the main results or findings.

3 habitats contribute to prokaryotic abundance on Earth.
  + Open ocean: 1.2x 10^29^ cells
  + Soil: 2.6x 10^29^ cells 
  + Subsurfaces: terrestrial and marine: 0.25-2.5x 10^30^ cells

Other important habitas:
  + Animals, leaves, air, groundwater
Further important habitats but with minor contributions to total cell number:  

Total number of prokaryotes on earth: 
  + ~4-6x 10^30^ cells

Total amount of Carbon in prokaryotes: 350-550 Pg which is roughly 60-100% of the total Carbon of plants

Total prokaryotic nutrients (N,P)= 10x nutrient content of plants. (N: 85-130 Pg, P: 9-14 Pg)

Turnover rates for the different habitats:  
  + soil: 2.5 years 
  + Ocean above 200m: 6-25 days  
  + Ocean below 200m: 300 days  
  + Terrestrial subsurface: 1-2x 10^3^ years

cellular productivity: # of cells produced per a unit of time: 
  + Ocean above 200m: highest productivity: 8.2*10^29^ cells/year
  + Total Earth: 1.7x 10^30^ cells/year

The higher the cellular productivity: the more mutation events occur: leads to more diversity. This leads to the emergence of new cycles and pathways.

Authors mention that the number of prokaryotic species may be underestimated due to inadequate sampling methods. 

#### Do new questions arise from the results?

* Subsurface prokaryote turnover rate: very high: is this an over-estimation: need to investigate this habitat further.

* Since the 1980s tools to study microbial diversity have evolved: do they give us better estimates for the above measurements?

* In the lsat 40 years, has the microbial abundance and diversity changed?

* How does the abundance of the prokaryotes in these environments play a role in the total metabolic capcity of these microbes in affecting the global nutrient cycles?

#### Were there any specific challenges or advantages in understanding the paper (*e.g.* did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)? 

It was hard to understand where some of the numbers, calculations and assumptions came from because it was not very well explained. A discussion on the precision of the data presented in this paper would have been cool.
 
\pagebreak


### Problem set 01

Paper: 
[Whitman *et al* 1998](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC33863/)


#### Learning objectives: 
Describe the numerical abundance of microbial life in relation to the ecology and biogeochemistry of Earth systems.

#### Specific questions:

#### What are the primary prokaryotic habitats on Earth and how do they vary with respect to their capacity to support life? Provide a breakdown of total cell abundance for each primary habitat from the tables provided in the text.  

Open ocean: Total 1.2x 10^29^ cells  

- Top 200m: 3.6x 10^28^  
- Below 200m (+10 cm of marine sediment): 8.2x 10^28^  

Soil: 2.5x 10^29^ cells

Subsurfaces (oceanic + terrestrial): ~3.8x 10^30^ cells

#### What is the estimated prokaryotic cell abundance in the upper 200 m of the ocean and what fraction of this biomass is represented by marine cyanobacterium including Prochlorococcus? What is the significance of this ratio with respect to carbon cycling in the ocean and the atmospheric composition of the Earth?  

Upper 200m of the ocean contains a total 3.6 x 10^28 cells:

->2.9x 10^27^ autotrophs (e.g, cyanobacteria like Prochlorococcus): 8% of total: responsible for carbon assimilation to sustain C requirements of heterotrophic (comprising 92% of total cells) cells. This menans that the carbon in the ocean is cycled within the ocean at higher rates than the carbon from the atmoshpere is fixed. This essentially results in a loss of carbon from the ocean into the atmosphere.

#### What is the difference between an autotroph, heterotroph, and a lithotroph based on information provided in the text? 

* Autotroph: Fix inorganic carbon (CO_2) -> biomass.   
* Heterotroph: use organic carbon to build biomass
* Lithotroph: inorganic substances as electron donor like NH3 

#### Based on information provided in the text and your knowledge of geography what is the deepest habitat capable of supporting prokaryotic life? What is the primary limiting factor at this depth?  

4km below the surface (4 km below terrestrial surface or marine sediments) is the deepest that life has been measured. The main limiting factor for life at this depth is temperature:
  * temperature is about 125 deg celcius: limit for prokaryotic life.
  * temperature drops at 22 degrees per km of sruface: Marian's trench, at 10.9 km deep, is the deepest part of the ocean that can support like: another 4 km below the marine sediments of Marian's trench would also contain life.

#### Based on information provided in the text your knowledge of geography what is the highest habitat capable of supporting prokaryotic life? What is the primary limiting factor at this height?

Spores can survivee at altitudes up to 77km above the surface of the earth. But metabollically active cells can only survive at roughly 20 km above the surface of the earth (so highest would be: Mt. everest: 8.8km + 20 km on top). Limiting factor: cold temperatures, ionizing radiationm lack of nutrients and no pressure.

#### Based on estimates of prokaryotic habitat limitation, what is the vertical distance of the Earth's biosphere measured in km?  

Mariana trench: ~ -10.9 km deep, + - 4 km further below marine sediment
Mt. everest: 8.8km + 20km on top 
vertical distance of Earth's biosphere: 
22 + 8.8 +10.95 +4= 45.7km

#### How was annual cellular production of prokaryotes described in Table 7 column four determined? (Provide an example of the calculation)

Annual cellular production
  * account for population size and growth rate
  * Annual cellular production = Population size (number of cells) x Turnover rate ( in years)
  * Using Subsurface ocean as an example:
      + # cells = 3.6x 10^28^ cells  
      + Turnover time: 16 days
      + Turnover rate = 365 (days/year)/16 ( turnover/days)= 22.81 turnovers per year
      + annual cellular production: 3.6x 10^28^ Cells x 22.81 turnovers per year = 8.2x 10^29^ cells per year
  * Soil:     
	    + # cell = 2.6x 10^29^ cells
	    + turnover time: 900 days
	    + Turnover rate = 0.4 turnovers per year (= 365/900)
	    + annual cellular production:2.6x 10^29^ x 0.4= 10^29^ cells per year

#### What is the relationship between carbon content, carbon assimilation efficiency and turnover rates in the upper 200m of the ocean? Why does this vary with depth in the ocean and between terrestrial and marine habitats?  

  * the net productivity in the ocean is: 51 Pg C/year
  * Total amount of marine C in ocean: 0.7-2.9 Pg
  * calculating the upper limit (2.9 Pg C)
      + Carbon in one cell: 5-20 fg C/cell
      + 20 fg C/cell=20x0^-30^ petagreams C
      + 3.6*10^28^ cells x 20x 10^-30^ pg/cell= 0.72 Pg of C in marine prokaryotes
      + Carbon efficieny while 20% (a factor 5 needed), Authors used 4 in the paper for some reason:
      + 4 x 0.72 Pg = 2.88Pg/year
  * 85% of net productivity in the ocean is consumed in the upper 200 m: 51 Pg C/year X 0.85 = ~43 Pg C/year
      + Turnovers per year in upper 200 m of ocean: 43 pg/yr / 2.88 pg/yr (turnover rate/total amounf of carbon in the ocean)= 14.9 Turnovers per year 
      + So: (14.9 turnovers/year) x (1 year/365 days) = one turnover every 24.5 days. or using the lower limit of C in the ocean: 61 turnovers per eyara ax.


The relationship between carbon assimilation efficiency adn turnover rate with depth varies in different environments  varies because the primary producer fraction of prokaryotes reach different depths in different habitats.

#### How were the frequency numbers for four simultaneous mutations in shared genes determined for marine heterotrophs and marine autotrophs given an average mutation rate of 4 x 10^-7^ per DNA replication? (Provide an example of the calculation with units. Hint: cell and generation cancel out)  

  * average mutation rate per DNA replication: 4 x 10^-7^ (mutations/gerneration)
  * for 4 simultaneous mutations: (4 x 10^-7^)^4^ = 2.56 x 10^-26^ mutations per gene per generation   
  *16 turnovers/day: 22.5turnovers/year
  * 3.6 x 10^28^cells (total number of marine cells) x 22.8 turnovers per year = 8.2 x 10^29^ cells produced per year    
  * (8.2 x 10^29^ cells per year) x (2.56 x 10^-26^ mutations per generation) = 2.1 x 10^4^  times per year 4 simulataneous mutations happen
      + 1/[(2.1 x 10^4^ events/ year) * (1 year/365 days) * (1 day /24 hours)]= 2.5 events/hour: or in other words
      + four simoultanous mutations happen every 0.4 hours every 24 minutes. 


#### Given the large population size and high mutation rate of prokaryotic cells, what are the implications with respect to genetic diversity and adaptive potential? Are point mutations the only way in which microbial genomes diversify and adapt?  

Due to high mutation rates and a large population size, the adaptive potential might be much higher than previously thought. The number of prokaryotic species might be much higher than previously estimated. The high diversity implies that prokaryotic cells have the potential to evolve and adapt relatively quick. High mutation rates might allow cells to adapt quickly to changing environments and new metabolic cycles can emerge that might have a great impact on the global nutrient cycle.

Beyond point mutations, microbes can diversify with bigger earrangements in the genome like inversions and deletions. Furthermore, genes can be adopted into the cells from DNA rom neighbouring cells or the environment in a process known as horizontal gene transfer.


#### What relationships can be inferred between prokaryotic abundance, diversity, and metabolic potential based on the information provided in the text? 

prokaryotic abundance correlates with magnitude of meetabolic potential.

prokaryotic diversity correlates with diversity of metabolic characteristics.

In conclusion: microbes have a large potential to participate in and alter the earth's biogeochemical cycles.



\pagebreak


### Evidence worksheet 02: Life and the Evolution of Earth's Atmosphere

Paper 1:
[Nisbet et al. 2001](https://www.nature.com/articles/35059210)
Paper 2:

[Kasting et al. 2002](http://science.sciencemag.org/content/296/5570/1066)

#### Learning objectives: 
Comment on the emergence of microbial life and the evolution of Earth systems

####Indicate the key events in the evolution of Earth systems at each approximate moment in the time series. 

**Hadean:**

* 4.6 Ga: 
    + Formation of solar system:
    + Earth: Atmosphere comprises of: water vapor, methane, ammonia, hydrogen, nitrogen.

* 4.5 Ga: 
    + Moon formed: gave Earth its spin, tilt, and day/ night cyles and seasons.

* 4.5 Ga - 4.1 Ga:
    + High levels of CO_2: increased temperature despite the sun being weak

* 4.4 Ga: 
    + Oldest zircoin forms
    + Oceans, and atmosphere forms  

* 4.4- 4.1 Ga: 
    + meteorite impacts

- 4.3-3.8 Ga: Heavy bombardment of Earth  

* 4.1 Ga: 
    + First evidence of life in zirocns and from preserved Carbon isotopes

* 4.0 Ga: Plate subduction  
    + Oldest rock: acasta gneiss in Canada and evidence of plate subduction.
    + CO_2 converted to carbonate: and lcts less as greenhouse gas: temperatures become glacial.
  

**Archaean:**  

* 3.8 Ga: 
    + Oldest sedimentary rocks show evidence of life and methanogenesis from carbon isotopes.
  
* 3.5 Ga: 
    + Evidence for photosynthesis: in fossils, and stromatolites.
    + Rubisco: enzyme for carbon assimilation detected: cyanobacteria photosynthesize: oxygen being produced.
  
* 3 Ga:
    + first Glaciation event: Oxygen production leads to less methanogen production.- less greenhouse effect  


* 2.7 Ga: 
    + great oxydation event-> glaciation 
    + Fossils contain molecules that suggest presence of cyanobacteria (hydrocarbon biomarkers) but also eukaryotes (steranes)
    
* 2.2 Ga:  
    + red rock beds obserevd: evidence of oxidation: oxygens level increased sharply.

**Proterozoic:**

* 1.8 Ga: 
    + Evidence of eukaryotes. 
    + Changing carbon cycle

* 1 Ga: 
    + First major ice age has a large impact on carbon cycle  

**Phanerozoic:**

* 540 Ma: 
    + Cambrian explosion: expansion of multicellular life
    + Emergence of animals.    

* 400 Ma: 
    + Land plants observed 
    + Strong increased oxygenation of atmosphere  
    + Denovian extinction: 

* 250 Ma: 
    + Permian extinction (95% of species disappear)  
    + gigantism of organisms, rise of dinosaurs  
    + Pangea forms, dry, harsh climate

* 66 Ma: 
    + Cretaceous-palogene extinction event
    + Dramatic global warming
    + Dominating forest  

* 200,000 years ago: 
    + *Homo sapiens* appears

#### Dominant physical/ chemical characteristics of Earth systems

**Hadean** 
  + High CO_2 levels (in the atmosphere) keep the earth warm: 500 deg Celcius
  + Loss of hydrogen to space
  + weak sun
  + Earth mostly molten rock
  + Rock vapour atmosphere
  + heavy meteror bombardment
  + Evidence of plate subduction near the end of the period
  
**Archean**  
  + Atmosphere failled with methane: keeps the earth warm: methanogenesis
  + Earth still anoxic at the beginning.
  + Earth starts cooling
  + Rock formation
  + As photosynthesis evolved, some O2 was present

**Proterozoic**
  + Oxygen levels in atmosphere rising
  + oxygen reacts with atmospheric methan to produce CO_2
  + atmospheric oxygen level rise resulting in glaciation events.
  + oxic air

**Phanerozoic**

  + Formation of pangea and subsequent breaking apart of pangea
  + Ice agaes
  + Modern atmosphere with increased oxygen levels.
  + Emergence of complex global nutrient cycles
  
-----

### Problem set 2: Microbial Engines

[Falkowski et al. 2008](https://www.ncbi.nlm.nih.gov/pubmed/18497287)

#### What are the primary geophysical and biogeochemical processes that create and sustain conditions for life on Earth? How do abiotic versus biotic processes vary with respect to mattern and energy transformation and how are they interconnected?  
    + Geophysical: plate tectonics, atmospheric photochemical processes: supply substrates and remove products: interaction of molecules + geochemical cycling.
    + Biogeochemical: Microbially catalyzed redox reactions  drove fluxes of major biolements.
    + Abiotic geochemical reactions are mostly based on acid/base chemistry (transfers of protons/H^+^), while biotic processes are based on redox reactions (successive transfers of electrons and protons). Abiotic reactions can can carry new molecules to biotic reactions supplying these reactions with new electrons for redox.

#### Why is Earth's redox state considered an emergent property?

The Earth's redox state depends on the biogeochemical cycles: feedback between between the evolution of microbial metabolica and geochemical processes is responsible for defining the average redox conditions of the oceans and atmosphere. An emergent property is something that as a whole is greater than the sum of its parts. The redox state of the planet of the Earth emerges from the combination of many different metabolic processes and reactions catalyzed by different organisms making it an emergent property.

#### How do reversible electron transfer reactions give rise to element and nutrient cycles at different ecological scales? What strategies do microbes use to overcome thermodynamic barriers to reversible electron flow?

A cycle, to be self-sustaning, must have both forward and reverse reactions. Communities of microbes living together interact with each through the transfer of elextrons via the exchange of metabolites, nutrients and wastes. The thermodynamic properties of a reaction is dependent on reaction conditions and organisms that contain thermodynamic conditions that are facored in these conditions can thrive. Synergistic cooperation between different species in the commpunity allow microbes to work together to overcome thermodynamic barriers to reversible electron flow. Reverse reractions can become possible when substrates become very low in concentrations. For example, the reduction of CO2 to methane using H_2 if the hydrogen pressure is low enough, the reverse reaction becomes thermodynamically favorable. The lowest H_2 concentration would potentially be adjacent to the H_2 utilizing, CO_2 reducing cells. On another scale, the reversible pathways may be catalyzed in a more global manner. 

#### Using information provided in the text, describe how the nitrogen cycle partitions between different redox "niches" and microbial groups. Is there a relationship between the nitrogen cycle and climate change?

Nitrogen fixation allows for the conversion of N~2~ into NH~4~ allowing the element to become accessible for anaerobic and metabolic processes. However, the enzymes responsible for this process is inhibited by O~2~. In the presence of O~2~, NH~4~ is oxidized to nitrate (thrugh nitrite) by a different set of nitrifying organisms. These nitrifiers concurrently perform these reactions to reduce CO~2~ into organic matter. In the absencce of oxygen, microbes may use NO2 and NO3 as electron acceptors in anaerobic oxidation which forms N~2~, thus clsing the nitrogen cycle. This closes the nitrogen cycle. Many different microbial groups are involved in different parts of the nitrogen cycle. 
Nitrifiers also reduce CO~2~ to organic matter. CO~2~ contributes to global warming and climate change. This links the nitrogen cycle to climate change.

#### What is the relationship between microbial diversity and metabolic diversity and how does this relate to the discovery of new protein families from microbial community genomes?

Fast turnover rates and large numbers of microbial cells leads to an enormous genetic diversity due to an accumulation of muutations and by acquisition of new genes through orizontal gene transfer. Differenet selective pressures exist in different environments which favor microbes that are capable of utilizing the available substrates. Specialized metabollic pathways arise in these environmetns as a result of the selective pressure and high genetic diversity. Throughout the world, in its myriad collection ofenvironments and selective pressures leads to huge metabolic diversity of microbes. Studies have found that the number of protein families within individual microbial genomes depedns linearly on the number of genes per genome.


#### On what basis do the authors consider microbes the guardians of metabolism?

Metabolic pathways in microbes have core genes like Rubisco in photoshynthesis. These genes are transferred bth vertically and horizontally to different microbes through strong environmental perturbations into various geological landscapes. Individual taxonomic groups may be sleced against but core metabolic machinery cotinues to survive through these environmental stresses. As a community, different organisms can provide different reactions that make-up a whole metabolic pathway. These different organisms are so well adapted to their environment, and so well distributed in different microbes that these metabolic pathways are unlikely to all go extinct. Even when less specialized microbes go extinct in a certain environment, the core metabolic gene set survives due to the varying envuronments and global distribution of enzymes.

\pagebreak


### Evidence Worksheet 03: Canfield DE et al. 2010. Science, Rockstrom J et al. 2009 Nature, OR Waters CN et al. 2016. Science

[Rockstrom et al. 2009](https://www.nature.com/articles/461472a)

#### Evaluate human impacts on the ecology and biogeochemistry of Earth systems

#### What were the main questions being asked?

Without pressure from humans, the Holocene, the period of geolical stabilty that we are experiencing now, is expected to continue for at least several thousand more years:
What are the planerary boundaries that must not be transgressed in order to prevent the irreversible damage to the earth's systems that keeps the planet in the Holcene?


#### What were the primary methodological approaches used?

defining planetary boundaries: safe operating space for humanity with respect to the Earth's biogeochemical and biophysical systems:
  + quantitaively defined as critical values for one or more control variables greater than which, human life and development become difficult to sustain (e.g. CO~2~ concentrations)
  + control variables that must have thresholds before unacceptable environmental changes soccur:
      + **Climate change**
          + boundary: containing rise in gobal mean temperature to no more than 2 deg C above the pre-inustrial level. based on 2 control variables:
                +  atmospheric CO~2~ concentrations
                + radiation: the rate of energy vchange per unit area of the globe as measured at the top of the atmosphere.
      + **Rate of bidiversity loss**
          + control variable: species loss
      + **interference with the N and P cycles**
          + control variable: rate of N2 removal from atmosphere and rate of N2 fixation.
      + stratospheric ozone depletion
      + ocean acidification
      + global freshwater use
      + Change in land use
      + Chemical pollution
      + Atmospheric aerosol loading
  +normative and conservative, risk averse approach to quantifying planetary boundary: taking into account large uncertainties that surround the true position of the threshold.
 

#### Summarize the main results or findings.

The boundaries of the safe operating space for humans has already been exceeded in three systems all due to human interference:
    + rate of biodiversity loss:
          + spcies are going extinct at a rate not seen since the lsat global mass-extinction event
          + Fossil data: 
          background extinction rate for marine life: 0.1-1 million species per year
          background extinction rate for mammals: 0.2-0.5 extinctions per million species per year.
          + estimated rate of extinction today: 100-1000x more than what would be considered normal (although this is a preliminary data as background extinction rate measurements aren't super confident): 30% mammals will be extinct by 2100 at this rate. Main reason: human activities like:
              + change in land use: urbanization and agriculture in natural ecosystems
              + Introduction of foreign species into new envionemts
              + change in distribution and patterns of wildfire occurance.
          + preserveration biodiversity linked with keeping ecosystems stable and resilient: the more redundnat and complicated a system: the more resilient it is
          + 
        
    + climate change:
          + CO~2~ levels should not exceed 350 ppmv (parts per million by volume) but current levels are at 387 ppmv 
          + radiative forcing should not exceed 1 watt per squre metere above pre-industrial levels but the change in radiative forcing is already 1.5 W/m^2^
          + Most models: a doubling of CO~2~ levles: increase global temperatures by 3 deg Celcius at equilibrium. 
              + but these models don't take into account the long term; postivie feedback loop propertie of global earming: warming leads to melting of icecdaps and changes in vegetation: slow feedbacks can result in an increase of 6 deg Celcius in stead of 3
              + warming interferes with ecological life-supporting systems in the evironemnt: would challenge the viability of contemporary human societies.
          + Some of earth's subsystems are already moving outside their stable Holcene state.
              + e.g. retreat of mounatin glaciers around the world, rise of sea-level in the last 2 decades and loss of global ice sheet mass. 
              
    + human interference with nitrogen cycle
        + causes:
            + Modern agriculture: abuse of N containing fertilizers: Human N fixation using artificial processes is more than the combined effort of **ALL** Earth's terrestrial processes. 
        + Addition of large amounts of usable N: significantly perturbs the global N cycle.Adding N: ends up in the environment and pollutes waterways an accumulating in land systems: adds a number of gasses to the atmosphere.
        + N2O^-^ the most important non Carbon dioxide greenhouse gas: adds to radiative forcing
        + control variable: N~2~ fixation from the atmosphere: should be 25% of what it is now.
        
Small changes in the Earth's biogeochemical and biophysical systems add up and through positive feedback loops: the situtation is exacerbated in the long term.

The different processes listed above are all interconnected in perturbation in one system is often accompanied by perturbations in other systems.

Fast approaching boundaries of three other Earth-system processes:
    + global freshwater use
    + change in land use
    + ocean acidification
    
Some of the data presented in the paper are preliminary results but the evidence so far suggests that remaining below the proposed thresholds for the above processes will allow humans the freedom to continue to pursue long-term social and economic development as they have been since the onset of the Industrial revolution.

#### Do new questions arise from the results?  

What kind of advice do the authors or other experts in the field hve to undo the damage that has already been done?

Are there any boundaries that we are unaware of that is being crossed?

In what ways can the metabolic capacity of microbes be harnessed to reverse some of these changes and to prevent further damage to the Earth's biogeochemial stability?


#### Were there any specific challenges or advantages in understanding the paper (e.g. did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)?
No real discussion of methods; it's more like a review.
It was easy to understand and the figures were easy to follow and really helped me understand the massive impact both human activities and microbial metabolism has on determining the Earth's biogeochemical and geological state.

\pagebreak

### Module 1 essay



### Module 01 references


Whitman WB, Coleman DC, and Wiebe WJ. 1998. Prokaryotes: The unseen majority. Proc Natl Acad Sci USA. 95(12):6578-6583.  
[Whitman et al. 1998](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC33863/)

Falkowski, P. G., T. Fenchel, and E. F. Delong. 2008. 'The microbial engines that drive Earth's biogeochemical cycles', Science, 320: 1034-9.  
[Falkowski et al. 2008](https://www.ncbi.nlm.nih.gov/pubmed/18497287)


Kasting, J. F., and J. L. Siefert. 2002. 'Life and the evolution of Earth's atmosphere', Science, 296: 1066-8.  
[Kasting et al. 2002](http://science.sciencemag.org/content/296/5570/1066)

Nisbet, E. G., and N. H. Sleep. 2001. 'The habitat and nature of early life', Nature, 409: 1083.  
[Nisbet et al. 2001](https://www.nature.com/articles/35059210)


Waters, Colin N., Jan Zalasiewicz, Colin Summerhayes, Anthony D. Barnosky, Clément Poirier, Agnieszka Galuszka,  
Alejandro Cearreta, Matt Edgeworth, Erle C. Ellis, Michael Ellis, Catherine Jeandel, Reinhold Leinfelder, J. R. McNeill, Daniel deB. Richter, Will Steffen, James Syvitski, Davor Vidas, Michael Wagreich, Mark Williams, An Zhisheng, Jacques Grinevald, Eric Odada, Naomi Oreskes, and Alexander P. Wolfe. 2016. 'The Anthropocene is functionally and stratigraphically distinct from the Holocene', Science, 351.  
[Waters et al. 2016](http://science.sciencemag.org/content/351/6269/aad2622)

[The Washington Post, 12-01-03, 'Spaceship Earth: A new view of envrionmentalism](https://www.washingtonpost.com/national/health-science/spacesh/)

[Canfield et al. 2010](http://science.sciencemag.org/content/330/6001/192)

A. Leopold, 1949, 'The Land Ethic', from 'A Sand County Almanac'

[Rockstrom et al. 2009](https://www.nature.com/articles/461472a)

D. P. Schrag, 2012, 'Geobiology of the Anthropocene'

Zehnder, 1988, 'Biology of anaerobic microorganisms', chapter 1







# Module 2

## Module 02 portfolio check

* Evidence worksheet_04
    + Completion status:
    + Comments:
* Problem Set_03
    + Completion status:
    + Comments:
* Supergraphic
    + Completion status:
    + Comments:
* Writting assesment_02
    + Completion status: N/A
    + Comments: cancelled
* Additional Readings
    + Completion status:
    + Comments:


### Problem set 03

Pre-reading:
[Wooley et al. 2010](http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1000667)
[Madsen et al. 2005](https://www.ncbi.nlm.nih.gov/pubmed/15864265)

#### How many prokaryotic divisions have been described and how many have no culturedrepresentatives (microbial dark matter)? 


[Solden *et al.* 2016](https://www.sciencedirect.com/science/article/pii/S1369527416300558): 89 bacterial phyla and 20 archaeal phyla (classified using SSU rRNA)
[Shloss et al 2004](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC539005/): 52 bacterial phyla: 26 of them uncultured
[Pace et al. 1997](https://www.ncbi.nlm.nih.gov/pubmed/9115194) : 40 bacterial divisions and 12 archael divsions
[Locey et al. 2016](http://www.pnas.org/content/113/21/5970): estimation using ecological scaling laws: 10^12^ bacterial species ont eh plannet
[Streit et al 2004](https://www.ncbi.nlm.nih.gov/pubmed/15451504) : estimated 99.8% bacteria on earth unculturable. 
Up to 99.8% of microbial species are not culturable (4).

#### How many metagenome sequencing projects are currently available in the public domain
and what types of environments are they sourced from?


[Gold database](https://gold.jgi.doe.gov/studies?Study.Metagenomic+Study=Yes&Study.Is+Public=Yes)
1424 metagenomic projects: and 51000 sequences
[EBI database](https://www.ebi.ac.uk/metagenomics/projects)
1500 metagenomic projects, and 111000 sequencing data sets stored online

Main environments to be sampled from are those that are hard to culture bactera from:
Air, soil, freshwater, marine, waste water, humans/animals, sediment

#### What types of on-line resources are available for warehousing and/or analyzingenvironmental sequence information (provide names, URLS and applications)?

Types of resources: metagenomic assembly, taxonomic binning, quality control, statistial analysis, online data bases to store data, as well as annotation.

  [EULER](https://omictools.com/euler-sr-tool)
  Metagenomic Assembly
  [MEGAN](http://ab.inf.uni-tuebingen.de/data/software/megan5/download/welcome.html)  
  Taxonomic binning
  [SILVA](https://www.arb-silva.de/)
  taxonomic binning based on 16s SSU rRNA
  [KEGG](http://www.genome.jp/kegg/) 
  Annotation, gene ontolog, metabollic pathways
  [MG_RAST](https://metagenomics.anl.gov/) 
  Gene assigning based on sequence similarity to database
  [AmpliconNoise](http://nebc.nox.ac.uk/bioinformatics/docs/Ampliconnoise.html)  
  quality filtering of sequences of bad quality
  [NCBI](http://ncbi.nlm.nih.gov/): 
  BLAST, sequencing tools, database to store and submit sequencing data, annotation
  [UniFrac](http://bmf.colorado.edu/unifrac)  
  comparing metagenomes based on phylogeny
  [EBI metagenoics](https://www.ebi.ac.uk/metagenomics/projects) taxonomic, functional and metagenomic analysis, database to store data.
  
sources:
Wooley et al. 2010 
[Dudhagara et. al, 2015](](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4678780/)

#### What is the difference between phylogenetic and functional gene anchors and how can they be used in metagenome analysis?

Phylogenetic anchors are usually genes that are transferred vertically from parent to progeny and are used to generate phylogenetic trees and to assign taxonomy to bacteria. An example are 16s SSu rRNA

Functional gene anchors are genes that are hallmarks of specific metaollic pathways and allow us to gain a better understanding of the metabollic capacity of the members in a microbial community.

[Barker et al. 2005](http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.0010003)

#### What is metagenomic sequence binning? What types of algorithmic approaches are used to produce sequence bins? What are some risks and opportunities associated with using sequence bins for metabolic reconstruction of uncultivated microorganisms?

metagenomic sequencing binning is the process involved in the association of sequening data to genes or taxonomy by building longer sequences from overlapping read.

there are two stategies to this:
composition-based binning and similarity-based binning:
composition based binning doesn't use a reference sequence to assemble metagenomes but instead uses the G/C content on frequency of codon usage to do this.

similarity based binning is dependent on taxonomy: reference sequences are used to assign homologs, like by using BLAST. Bias towards known sequences. this method aloows for phylogenetic classification.

the risks of using sequence bins for metabolic reconstruction are many: compositon based-binning is often eror-prone especially when closely related OTUs are seen.
Often whole genomes are not covered in the metagenomic sequencing so it is hard to completely talk about the metabolic capacity of different OTUs. incomplete coverage of metagenomes also leads to an incrase in inter-species chimera formation. Also, DNA from the species present at lower abundances is not detected as easily. Some sequences cannot be assigned because they do not match what is already known in the databases.

Opportunities: Get a broad overview of the metagenomic potential of a community. it is fast, efficient and based on algorithms that are constantly improving. These methods are definitely more insightful than culture dependent methods.  

#### Is there an alternative to metagenomic shotgun sequencing that can be used to access the metabolic potential of uncultivated microorganisms? What are some risks and opportunities associated with this alternative?
 

Single Cell sequencing:
expensive, and takes a long time. Algorithms associated with this are still being developed. not amplification based so hard to weed out sequencing errors.

Third generation sequencing: Oxford Nanopre, PacBio
No amplification needed, long reads and fast but high error rates at this point and concentrated samples are needed.

Functional screens: biochemical
not high-throuput but can directly asses metabollic potetnial of a comunity by testinf the ability to utilize specific substrates.

Fluoresence In Situ Hybridization:
can look for specific sequence existence in a community: have to look at highly conserved markers
Other methods:   

Source: Wooley et al, 2010


\pagebreak

### Evidence worksheet 04
Reading:
[Martinez et al. 2007](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1838496/)

#### What were the main questions being asked?  

Characterizing Photorhodopsin photosystems: structure, function and genetics
What are the minimal genetic elements required to compement the photorhodopsin system into E. coli. 


#### What were the primary methodological approaches used?  
DNA from environment (marine microplankton) -> F plasmid with increased duplication using an L-arabinose induced ORI -> transform into E. coli, look for orange growth on retinal containing and lacking media.-> isolate vectors: transposon mutagenesis -> transform into E. coli again: look for loss of phenotype: transposon sequening of loci with inserted sequence.
Light activated PR proton transfer measured using pH in media containing PR containing and PR deficient colonies. ATP synthesis was also measured using controls and using molecules to disrupt ATP synthase or to disrupt mitochondrial membrane impermeability to H+.

#### Summarize the main results or findings.  

-Engineering of hetrologous microbes!!!
-PR based photosystems can be expressed and function in E. coli without the addition of exogenous retinal. found an operon containing 6 genes involved in beta-carotene and retinal biosyntehsis which was adjacent to the photorhodopsin gene by sequencing. These genes are closely rranged and likely transfer together to other microbes. in E. coli the PR systems are capable of generating a proton gradient potentriating these cells to generate ATP using light (phototrophism). Also seen activation of phosphorylation of ADP by ATP synthase. High abundance of this system in marine environments indicate high levels of horizontal gene transfer.

Also cool: increasing F plasmid number of a single plasmid in bacteria: enhanegene expression: make it easier to screen for + and - phenotpyes.

#### Do new questions arise from the results?

- photorhodopsin and enhanced growth have not been definitively linked but so many microes have this system.

-What other functions might the photo-rhodopsin have in these marine bacteria othere than phototrophy: chemosensory, flagellar motility, nutrient acquisition functions


#### Were there any specific challenges or advantages in understanding the paper (e.g. did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)?  

long and tons of genetic jargon but overall easy to read.

### Module 2 References


# Module 3

## Module 03 portfolio check

* Problem Set_04
    + Completion status:
    + Comments:
* Evidence worksheet_05
    + Completion status:
    + Comments:
* Writing assessment_03
    + Completion status:
    + Comments:
* Additional Readings
    + Completion status:
    + Comments  


### Problem set 04: Fine-scale phylogenetic architecture

#### Learning objectives: 

* Gain experience estimating diversity within a hypothetical microbial community


#### Outline:

In class Day 1:

1. Define and describe species within your group's "microbial" community.
2. Count and record individuals within your defined species groups.
3. Remix all species together to reform the original community.
4. Each person in your group takes a random sample of the community (*i.e.* devide up the candy).

Assignment:

5. Individually, complete a collection curve for your sample.
6. Calculate alpha-diversity based on your original total community and your individual sample.

In class Day 2:

7. Compare diversity between groups.

#### Part 1: Description and enumeration
Loading in the packages that will be used.
```{r message=FALSE}
#To make tables
library(kableExtra)
library(knitr)
#To manipulate and plot data
library(tidyverse)
#Alpha-diversity functions in R
library(vegan)
```

loading Data:
```{r}
candy_data = data.frame(
  number = c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52),
  name = c("M&Ms_Brown","M&Ms_Red","M&Ms_Blue", "M&Ms_Yellow", "M&Ms_Orange", "MM&Ms_Green","Skittles_Yellow"," â€œSkittles_Purple", "Skittles_Orange", "Skittles_Green", "Skittles_Red", "Big_circle_Orange", "Big_circle_Green", "Big_circle_Yellow","Big_circle_Red", "Big_circle_Purple", "Red_Strings", "Jolly_Ranchers_Yellow", "Jolly_Ranchers_Green", "Jolly_Ranchers_Orange", "Jolly_Ranchers_Pink", "Jolly_Ranchers_Red", "Hershies","Legos_BlueRectangles", "Legos_BlueSquares", "Legos_PinkRectangles", "Legos_PinkSquares", "Legos_YellowRectangle", "Legos_YellowSquare", "Legos_GreenRectangles", "Wine_gummies_Purple", "Wine_gummies_Orange", "Wine_gummies_Red", "Wine_gummies_Yellow", "Octopus_Orange/Green", "Octopus_Yellow/Pink", "Octopus_Green_Pink", "Octopus_Purple/Yellow", "Swirl_Red", "Swirl_Blue", "Weird_gummies_cola", "Weird_gummies_Cherry", "Weird_gummies_Wine", "Classic_Bear_Red", "Classic_Bear_Purple", "Classic_Bear_Orange", "Classic_Bear_White", "Classic_Bear_Yellow", "Classic_Bear_Green", "sugar_coated_gummies_Red", "sugar_coated_gummies_Pink", "sugar_coated_gummies_Yellow"),
  occurences = c(29,27,64,35,59,28,31,43,42,46,35,5,5,4,2,3,14,27,33,44,38,33,16,3,1,6,1,3,1,2,1,1,2,3,2,2,1,1,1,2,3,1,2,16,16,16,16,18,19,1,1,1)
)
```

Creating a Data Table
```{r}
candy_data %>% 
  kable("html") %>%
  kable_styling(bootstrap_options = "striped", font_size = 10, full_width = F)
```
	

#### Part 2: Creating a Collector's curve

Loading in my sample data.
```{r}
my_sample = data.frame(
 x=c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151),
  y=c(1,1,1,1,1,1,1,1,1,2,2,2,2,2,2,2,3,3,3,3,3,3,3,3,3,3,3,3,4,4,4,5,5,5,5,5,5,5,5,5,5,5,6,6,6,6,6,7,7,7,7,7,7,7,7,8,8,8,8,8,8,8,9,9,9,9,9,9,9,9,9,9,10,10,10,10,10,10,10,11,11,11,11,11,11,11,11,12,13,14,14,14,14,15,15,15,16,16,16,16,16,16,17,17,17,17,17,17,17,17,17,17,18,18,18,18,18,19,19,19,19,20,20,20,20,21,22,23,24,25,26,27,28,29,29,29,29,29,29,30,30,30,30,31,31,32,32,32,32,33,33)

)
```

Creating the plot.
```{r}
ggplot(my_sample, aes(x=x, y=y)) +
  geom_point() +
  geom_smooth() +
  labs(x="Cumulative number of individuals classified", y="Cumulative number of species observed")
```

* Does the curve flatten out? If so, after how many individual cells have been collected?

The curve produced continues to increase sharply as more cells are classified.

* What can you conclude from the shape of your collector's curve as to your depth of sampling?

The cap on number of species observed is limited by the sampling size: only about 2/3 of all species have been observed from this small sample

#### Part 3: Diversity estimates (alpha diversity)

##### Diversity: Simpson Reciprocal Index

$\frac{1}{D}$ where $D = \sum p_i^2$

$p_i$ = the fractional abundance of the $i^{th}$ species

For example, using the example data 1 with 3 species with 2, 4, and 1 individuals each, D =

```{r}
species1 = 29/(806)
species2 = 27/(806)
species3 = 64/(806)
species4=35/(806)
species5=59/(806)
species6=28/(806)
species7=31/(806)
species8=43/(806)
species9=42/(806)
species10=46/(806)
species11=35/(806)
species12=5/(806)
species13=5/(806)
species14=4/(806)
species15=2/(806)
species16=3/(806)
species17=14/(806)
species18=27/(806)
species19=33/(806)
species20=44/(806)
species21=38/(806)
species22=33/(806)
species23=16/(806)
species24=3/(806)
species25=1/(806)
species26=6/(806)
species27=1/(806)
species28=3/(806)
species29=1/(806)
species30=2/(806)
species31=1/(806)
species32=1/(806)
species33=2/(806)
species34=3/(806)
species35=2/(806)
species36=2/(806)
species37=1/(806)
species38=1/(806)
species39=1/(806)
species40=2/(806)
species41=3/(806)
species42=1/(806)
species43=2/(806)
species44=16/(806)
species45=16/(806)
species46=16/(806)
species47=16/(806)
species48=18/(806)
species49=19/(806)
species50=1/(806)
species51=1/(806)
species52=1/(806)

1 / (species1^2 + species2^2 + species3^2 + species4^2 + species5^2 + species6^2 + species7^2 + species8^2 + species9^2 + species10^2 + species11^2 + species12^2 + species13^2 + species14^2 + species15^2 + species16^2 + species17^2 + species18^2 + species19^2 + species20^2 + species21^2 + species22^2 + species23^2 + species24^2 + species25^2 + species26^2 + species27^2 + species28^2 + species29^2 + species30^2 + species31^2 + species32^2 + species33^2 + species34^2 + species35^2 + species36^2 + species37^2 + species38^2 + species39^2 + species40^2 + species41^2 + species42^2 + species43^2 + species44^2 + species45^2 + species46^2 + species47^2 + species48^2 + species49^2 + species50^2 + species51^2 + species52^2)

s1= 9/(151)
s2= 7/(151)
s3= 12/(151)
s4= 3/(151)
s5= 11/(151)
s6= 5/(151)
s7= 7/(151)
s8= 10/(151)
s9= 7/(151)
s10 =8/(151)
s11 =1/(151)
s12 =1/(151)
s13= 4/(151)
s14= 3/(151)
s15= 6/(151)
s16= 10/(151)
s17= 5/(151)
s18= 4/(151)
s19= 4/(151)
s20= 1/(151)
s21= 1/(151)
s22= 1/(151)
s23= 1/(151)
s24= 1/(151)
s25= 1/(151)
s26= 1/(151)
s27= 1/(151)
s28= 8/(151)
s29= 6/(151)
s30= 4/(151)
s31= 2/(151)
s32= 4/(151)
s33= 2/(151)
  
1/((s1^2) + (s2^2) +(s3^2) + (s4^2) +(s5^2) + (s6^2) +(s7^2) + (s8^2) +(s9^2) + (s10^2) + (s11^2) + (s12^2) +(s13^2) + (s14^2) +(s15^2) + (s16^2) +(s17^2) + (s18^2) +(s19^2) + (s20^2) + (s21^2) + (s22^2) +(s23^2) + (s24^2) +(s25^2) + (s26^2) +(s27^2) + (s28^2) +(s29^2) + (s30^2)+ (s31^2) + (s32^2) +(s33^2))



```


* What is the Simpson Reciprocal Index for your sample? 21.53069
* What is the Simpson Reciprocal Index for your original total community? 23.44748

##### Richness: Chao1 richness estimator

$S_{chao1} = S_{obs} + \frac{a^2}{2b})$

$S_{obs}$ = total number of species observed
a = species observed once
b = species observed twice or more


```{r}
#my community: 
52 + 12^2/(2*40)

#my sample: 
33 + 10^2/(2*23)
```

* What is the chao1 estimate for your sample? 35.17391
* What is the chao1 estimate for your original total community? 53.8

#### Part 4: Alpha-diversity functions in R

Removing the unnecesary data columns and transposing the data so that `vegan` reads it as a species table with species as columns and rows as samples.
```{r}
candy_data1_diversity = 
  candy_data %>% 
  select(name, occurences) %>% 
  spread(name, occurences)

candy_data1_diversity
```

Calculating the Simpson Reciprocal Index using the `diversity` function.
```{r}
diversity(candy_data1_diversity, index="invsimpson")
#23.44748
```

Calculating the Chao1 richness estimator (and others by default) with the the `specpool` function for extrapolated species richness.
```{r}
specpool(candy_data1_diversity)
#    Species chao chao.se jack1 jack1.se jack2 boot boot.se n
#All      52   52       0    52        0    52   52       0 1
```

repeating the calculations for my sample:
```{r}
my_sample_data = data.frame(
  number = c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33),
  name = c("M&Ms_Brown","M&Ms_Red", "M&Ms_Blue", "M&Ms_Yellow", "M&Ms_Orange", "MM&Ms_Green","Skittles_Yellow"," â€œSkittles_Purple", "Skittles_Orange", "Skittles_Green", "Skittles_Red", "Big_circle_Green", "Big_circle_Yellow","Red_Strings", "Jolly_Ranchers_Yellow", "Jolly_Ranchers_Green", "Jolly_Ranchers_Orange", "Jolly_Ranchers_Pink", "Jolly_Ranchers_Red", "Hershies", "Legos_BlueSquares", "Legos_PinkRectangles", "Legos_PinkSquares", "Legos_GreenRectangles", "Wine_gummies_Orange", "Wine_gummies_Yellow", "Octopus_Purple/Yellow", "Swirl_Red", "Classic_Bear_Red", "Classic_Bear_Orange", "Classic_Bear_White", "Classic_Bear_Yellow", "Classic_Bear_Green"),
  occurences = c(9,7,12,3,11,5,8,7,10,7,8,1,1,4,3,6,10,5,4,4,1,1,1,1,1,1,1,1,6,4,2,4,2)
)
my_sample_data %>% 
  kable("html") %>%
  kable_styling(bootstrap_options = "striped", font_size = 10, full_width = F)

my_sample_data1_diversity = 
  my_sample_data %>% 
  select(name, occurences) %>% 
  spread(name, occurences)

my_sample_data1_diversity

diversity(my_sample_data1_diversity, index="invsimpson")
#21.53069

specpool(my_sample_data1_diversity)
#    Species chao chao.se jack1 jack1.se jack2 boot boot.se n
#All      33   33       0    33        0    33   33       0 1

```


In Project 1, you will also see functions for calculating alpha-diversity in the `phyloseq` package since we will be working with data in that form.

For your sample:

* What are the Simpson Reciprocal Indices for your sample and community using the R function?
  community: 23.44748
  my Sample: 21.53069
* What are the chao1 estimates for your sample and community using the R function?
  community:52
  my sample:33
    + Verify that these values match your previous calculations: the Simpson Reciprocal Indices are exactly the same, and the chao1 estimates are relatively close.

#### Part 5: Concluding activity

* How does the measure of diversity depend on the definition of species in your samples?

The less stringent your classification scheme, the less species diversity will be seem. If different species definitions lead to the same number of species but different abundance:diversity ratios among the species: this will also affect the diversity calculations


* Can you think of alternative ways to cluster or bin your data that might change the observed number of species?

We could have disregarded colour in our classification scheme: this would drastially reduce the number of species observed and have an effect on diversity calculations

* How might different sequencing technologies influence observed diversity in a sample?
Sequencing methods that are error prone like the oxford nan-pore or sanger-sequencing: the observed diversity will likely be overestimated or you would get misclassification of some of our OTUs. Furthermore, more errors in sequencing platforms lead to more diversity in bp in the data collected resulting in an increased amount of diversity in the sample.


### Evidence Worksheet 05: Extensive mosaic structure

[Welch et al. 2002]( http://www.pnas.org/content/99/26/17020)

#### What were the main questions being asked?
-How do the genomes of different E. coli differ (specifically 3 closely related strains)and how do these differences play a role in the pathogenicity of E. coli strains.
-Comparing genome structures of E. coli to understand evolution of E. coli.
  


#### What were the primary methodological approaches used?

Genome sequence of:
uropathogenic E. coli strain CFT073 (from a woman with acute pylonephritis) compared to
enterohemorrhagic strain EDL933 and 
nonpathogenic lab strain MG1655.    

Genome sequences analyzed using MAGPIE, and homologs with the other 2 strains were searched for using BLAST: genes with 90% similarity were determined to be orthologues.

#### Summarize the main results or findings. 

-	The CFT073 genome sequenced and a restriction enzyme map was generated. The genome had no plasmids and had roughly 5500 coding genes. CTF073 has more pathogenicity islands and virulence gens than MG6155 (1.303 Mb!! vs 716 kb). The ori and terminuts of replication between the two strains was identical. In CFT073: the codons used in the pathogenicity island differed from the rest of the gene: indication of horizontal gene transer. 

CFT07 has no plasmids and type 3 secretion system while EDL933 does: can give insight into how the pathogenicity of these 2 strains differs. CFT07 genome has fimbial adhesins, recombinases, autotransporters for exporting virulence factors, hemolysi genes that encode cytolytic toxins and contains a pathogenicity island from Yersinia pestis.
roughly 40% of all proteins were shared between all 3 strains and CFT07 had 20% of its genome being exclusive to itself. the rest of the genome was shared between only 2 of the 3 strains.
Pathogeneic strains definitely more genetially diverse than lab strains because of genes that they pick up from the environment during their foray into the harsh and selective environment of the intestine or urinary regions.
 The location of the plasmids or pathogenicity islands in the genome of these strains helps define which strain they are.

#### Do new questions arise from the results?

What gene subsets differentiate E. coli that have tissue tropism for different cell types in the himan body?

Does the high sequence similarity critieria become obsolete when talking about assigning taxonomy to species with increasing amounts of horizontal gene transfer as these strains have very different functional capacities from each other?


#### Were there any specific challenges or advantages in understanding the paper (e.g. did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)? 

The method section was not very detailed but in general the paper was okay. 

#### Based on your reading and discussion notes, explain the meaning and content of the following figure derived from the comparative genomic analysis of three E. coli genomes by Welch et al. Remember that CFT073 is a uropathogenic strain and that EDL933 is an enterohemorrhagic strain. Explain how this study relates to your understanding of ecotype diversity. Provide a definition of ecotype in the context of the human body. Explain why certain subsets of genes in CFT073 provide adaptive traits under your ecological model and speculate on their mode of vertical descent or gene transfer.

![](c:/Users/Jawairia/Documents/MICB425_portfolio/Evidence_worksheet5_image.jpg)


An ectotype is a bacterial strain that is well adapted to a specific ecological niche and is unlikely to be found elsewhere. In other words, it is a genetically distinct geographic variety of organism within a species.In the human body, these geographic locations can be different physical locations like the small intestine, the inguinal region or the urinary tract. Bacteria in these diffrent locations face different selective pressures based on the environmal conditions that are present. Even on a microscopic level, for example, in the gut, bacteria in the center of the intestinal lumen and those closely associated with the microvilli face different pressures. These different ecotypes all share 16s SSU rRNA sequences but must be able to survive different environments and are thus differentially transcriptically active. As a result, different ecotypes can differ significantly in the genetic islands specific for the niche that they occupy.

The figure depicts the different genetic islands in the genomes of the two human pathogenic strains of E. coli. These islands were likely selected for by the different environments and nutrients these bacteria encounter. The backbone of the genome is shared between the two E. coli strains. The backbone genes are likely transfered vertically from parent to progeny but the differential presence of the genetic islands is due to horizontal gene transfer. This leads me to believe that horizontal gene transfer is most likely the key process behind the evolution of ecotypes.


### Module 3 essay

Discuss the challenges involved in defining a microbial species and how HGT complicates matters, especially in the context of the evolution and phylogenetic distribution of microbial metabolic pathways. Can you comment on how HGT influences the maintenance of global biogeochemical cycles through time? Finally, do you think it is necessary to have a clear definition of a microbial species? Why or why not?



How to decide for a clear definition of microbial species is an often-discussed issue. The biological species concept cannot be applied for microbes as they do not reproduce sexually. Many alternative concepts exist that use different approaches to define species. Each of these definitions has their own advantages and disadvantages, which makes it difficult to agree on one that fits the needs of all purposes. Some of the most important concepts will be discussed in more detail in this assay. At first, probably the most often applied concept is to assign two individuals to the same operational taxonomic unit (OTU) if their 16s rDNA shows more than 97% identity. A more recently developed method is to define species by amplicon sequence variants (ASV). This method has the potential of even higher resolution than the OTU definition as it defines an exact nucleotide sequence for each species. Another factor that makes species definitions even more challenging is horizontal gene transfer (HGT). This microbial version of "sexual" recombination leads to difficulties in many phylogenetic approaches. Through HGT, even distantly related species can harbor very similar DNA fragments. This allows for a global distribution and maintenance of key metabolic pathways but in turn also leads to a high potential for wrong species-assignments. One method to overcome the problems of HGT is multi-locus sequence analysis (MLSA). MLSA tries to eliminate genes acquired from horizontal transfer by comparing several conserved genes instead of just one, as the OTU and ASV methods do. Regarding all these challenges in species definition, the question arises whether it is even necessary to have a clear definition? In order to find an answer to this question, it is necessary to first obtain a deeper understanding of the present species definitions.  


Defining microbial species by a 16s RNA identity of 97% or more has become the standard approach over the last years. This method has been shown to achieve well dissolved phylogenetic assignments in numerous cases. However, with increasing amounts of available data, several problems of this method could be identified. Firstly, as only the 16s rDNA is considered, a lot of variation might be missed. Two species can have completely different sequences in the main parts of their genomes, with just very similar 16s rDNA and thus get clustered together as the same species. Second, one species can have multiple copies of the 16s RNA gene. This means that we have to decide which of the copies we want to use for the species definition. In addition, if the sequence data comes from an unknown environmental sample, one might not even realize that the two 16s rDNA sequences come from the same genome and therefore assign them to two different species. Third, as more genomic data of very closely related species became available, the resolution of 16s rDNA turned out not to be sufficient anymore to separate these species. In these cases, the 16s rDNA shows too high levels of conservation to be sufficient for separation of very similar species. Another challenge that arises from this problem is the decision of how to differentiate between species and strains. This problem again could be the subject of a whole essay.  
	
	
The more recent amplicon sequence variants (ASV) method has been shown to overcome some of the problems of the 97% rDNA identity OTU definition approach. This method does not rely on an arbitrarily chosen identity threshold, but instead defines species by one exact nucleotide sequence[1]. Therefore, this method provides higher resolution as it distinguishes sequences that even differ by just one nucleotide. However, this method too struggles with some challenges in species definition. In the process of defining one exact sequence to be the true one, many sequences which are assumed to contain sequencing errors have to be discarded. This means, that sequences which are more error-prone get discarded more often and might not be detected as species. In addition, this denoising step is very laborious and therefore requires a lot of capacity if big data sets have to be processed. Another problem is that if a wrong sequence is chosen as a species-defining ASV, the mistake immediately leads to an incorrectly defined species. One challenge the ASV method shares with the OTU method is, that it tries to define species by only one DNA locus. Thus, with ASV again, there is no guarantee that two species actually are the same or even closely related just because they have the same sequence in the observed amplicon.   
	
	
The past two paragraphs show that the process of defining species can be very challenging. One factor that makes these definitions even more difficult is horizontal gene transfer (HGT). Many bacterial species can exchange some of their genes through HGT even if they are very distantly related. Therefore, individual genes within a single genome can have different evolutionary histories. This leads to varying phylogenetic trees and species assignments, depending on which genes are analyzed. For example, two species can have high similarity in 16s rDNA and other housekeeping genes, but still differ in great parts of their genomes, like specific pathogenicity islands[2]. Else, due to HGT, two species can be identical in most parts of their genomes, with differences in only a few very specific loci [3]. In both cases, 16s rDNA alone would not be sufficient to separate the ecotypes that share the same ribosomal DNA but have clear phenotypic or ecological differences in terms of the niches they occupy. A method to deal with the problem of HGT interfering with the definition of these ecotypes, is multi-locus sequence analysis (MLSA). In MLSA, several conserved housekeeping genes are concatenated and analyzed [4]. This method has two major positive properties. First, the interfering effects of HGT get eliminated, with increasing success, the more different genes are analyzed. Second, using multiple genes for sequence comparison, strongly increases the obtained resolution to distinguish species or ecotypes. Thus, the MLSA method is well worth considering as alternative to the single gene based methods which are more prone to get biased by HGT.   
	
	
Because of horizontal gene transfer and following selective pressure, the genes for many key metabolic pathways get distributed all over the world in many different species. On one hand, this makes it difficult to generate phylogenetic trees and distinguish species, as discussed in the previous paragraph. But on the other hand, HGT assures the maintenance of global biogeochemical cycles, allowing the microbes to function as guardians of the metabolism [5]. Different environments with strong selective pressures favor the survival of only the best adapted microbes. Bacteria that fail to adapt fast enough to a changing environment go extinct. However, because of HGT, their specialized metabolic pathways often survive in another species in a different environment somewhere else on the planet. Thus, HGT ensures the maintenance of global biogeochemical cycles through time as it enables the distribution of key metabolic genes across species all over the world.  
  
  
The massive amount of HGT plays an important role in the distribution of metabolic pathways and evolution of microbes. This extensive genomic fluidity strongly increases the challenges in defining microbial species and has led to some scientists questioning the existence of any microbial species [6]. But is it even necessary to have a clear definition of microbial species? Intuitively, one might think that a single, universal definition of microbial species would be very useful. Assigning an individual to a species would be much simpler if just one clear rule with a universal cutoff existed. Also, if two species are compared, one would immediately know the minimal level of genetic difference between the two species. However, despite these obvious and strong advantages of a clear definition, I think that it is important to allow for varying species definitions. An example from Fraser et al [7] illustrates this on three closely related bacterial species: Streptococcus pneumoniae, a human pathogen, S. mitis, a commensal and S. pseudopneumoniae, a species of uncertain status. Of these three species, S. mitis contains the highest genetic diversity of about 5%. This distance is about the same as the average distance between two genomes of S. pneumoniae and S. pseudopneumoniae. Therefore, applying a universal cutoff means that either S. pneumoniae and S. pseudopneumoniae had to be clustered together, or S. mitis would be split into many distinct species. This example clearly shows that one clear species definition with a defined cutoff is not only unnecessary, but even detrimental in many cases. Often, several context dependent aspects have to be considered in order to achieve satisfying clustering of microbes. For the generation of phylogenetic trees, varying phenotypical and ecological properties are arguably as important as genetic similarity. By considering only the 16s rDNA identity, many ecotypes would be missed in the analysis. Thus, it is often important to look at many different metabolic genes to be able to identify differences in ecotypes occupying different niches, despite their great genetic similarity.  
  
  
In conclusion, the definition of microbial species encounters many challenges. Several approaches exist that try to sort species by a clearly defined threshold of genetic similarity in some single or multiple genes. Each of these methods has their own advantages and disadvantages and none can be clearly considered as the best one. A factor making species definitions even more complicated is HGT. Extensive gene exchanges distribute key genes all across the bacterial domain, leading to the maintenance of biogeochemical cycles by a globally shared core genome. This genome fluidity allows for fast bacterial evolution and adaptation, but in turn causes serious problems in the attempt of defining species. However, a clear species definition is not always necessary after all. Satisfying clustering of microbes often needs context dependent, varying definitions, considering also ecological aspects and many metabolic genes, not just similarity in the 16s rDNA. As a new approach, it might even be worth considering to completely give up on any universal species concepts. Instead, many different methods could be applied simultaneously, leading to several different, even overlapping clusters of taxa. Therefore, for each application, the best clustering method could be chosen to address the particular problem or question in an optimal way.


####References  

1.	Callahan, B.J., P.J. McMurdie, and S.P. Holmes, Exact sequence variants should replace operational taxonomic units in marker-gene data analysis. ISME J, 2017. 11(12): p. 2639-2643.
2.	Welch, R.A., et al., Extensive mosaic structure revealed by the complete genome sequence of uropathogenic <em>Escherichia coli</em>. Proceedings of the National Academy of Sciences, 2002. 99(26): p. 17020-17024.
3.	Shapiro, B.J., et al., Population genomics of early events in the ecological differentiation of bacteria. Science, 2012. 336(6077): p. 48-51.
4.	Glaeser, S.P. and P. Kampfer, Multilocus sequence analysis (MLSA) in prokaryotic taxonomy. Syst Appl Microbiol, 2015. 38(4): p. 237-45.
5.	Falkowski, P.G., T. Fenchel, and E.F. Delong, The microbial engines that drive Earth's biogeochemical cycles. Science, 2008. 320(5879): p. 1034-9.
6.	Riley, M.A. and M. Lizotte-Waniewski, Population Genomics and the Bacterial Species Concept. Methods in molecular biology (Clifton, N.J.), 2009. 532: p. 367-377.
7.	Fraser, C., et al., The Bacterial Species Challenge: Making Sense of Genetic and Ecological Diversity. Science, 2009. 323(5915): p. 741-746.




### Module 3 References

