---
title: "Problem Set_04"
author: "Jawairia Atif (39139134)"
date: "version `r format(Sys.time(), '%B %d, %Y')`"
output: html_document
editor_options: 
  chunk_output_type: console
---
### Problem set_04 “Fine-scale phylogenetic architecture”
#### Learning objectives: 

* Gain experience estimating diversity within a hypothetical microbial community


#### Outline:

In class Day 1:

1. Define and describe species within your group's "microbial" community.
2. Count and record individuals within your defined species groups.
3. Remix all species together to reform the original community.
4. Each person in your group takes a random sample of the community (*i.e.* devide up the candy).

Assignment:

5. Individually, complete a collection curve for your sample.
6. Calculate alpha-diversity based on your original total community and your individual sample.

In class Day 2:

7. Compare diversity between groups.

#### Part 1: Description and enumeration
Loading in the packages that will be used.
```{r message=FALSE}
#To make tables
library(kableExtra)
library(knitr)
#To manipulate and plot data
library(tidyverse)
#Alpha-diversity functions in R
library(vegan)
```

loading Data:
```{r}
candy_data = data.frame(
  number = c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52),
  name = c("M&Ms_Brown","M&Ms_Red","M&Ms_Blue", "M&Ms_Yellow", "M&Ms_Orange", "MM&Ms_Green","Skittles_Yellow"," “Skittles_Purple", "Skittles_Orange", "Skittles_Green", "Skittles_Red", "Big_circle_Orange", "Big_circle_Green", "Big_circle_Yellow","Big_circle_Red", "Big_circle_Purple", "Red_Strings", "Jolly_Ranchers_Yellow", "Jolly_Ranchers_Green", "Jolly_Ranchers_Orange", "Jolly_Ranchers_Pink", "Jolly_Ranchers_Red", "Hershies","Legos_BlueRectangles", "Legos_BlueSquares", "Legos_PinkRectangles", "Legos_PinkSquares", "Legos_YellowRectangle", "Legos_YellowSquare", "Legos_GreenRectangles", "Wine_gummies_Purple", "Wine_gummies_Orange", "Wine_gummies_Red", "Wine_gummies_Yellow", "Octopus_Orange/Green", "Octopus_Yellow/Pink", "Octopus_Green_Pink", "Octopus_Purple/Yellow", "Swirl_Red", "Swirl_Blue", "Weird_gummies_cola", "Weird_gummies_Cherry", "Weird_gummies_Wine", "Classic_Bear_Red", "Classic_Bear_Purple", "Classic_Bear_Orange", "Classic_Bear_White", "Classic_Bear_Yellow", "Classic_Bear_Green", "sugar_coated_gummies_Red", "sugar_coated_gummies_Pink", "sugar_coated_gummies_Yellow"),
  occurences = c(29,27,64,35,59,28,31,43,42,46,35,5,5,4,2,3,14,27,33,44,38,33,16,3,1,6,1,3,1,2,1,1,2,3,2,2,1,1,1,2,3,1,2,16,16,16,16,18,19,1,1,1)
)
```

Creating a Data Table
```{r}
candy_data %>% 
  kable("html") %>%
  kable_styling(bootstrap_options = "striped", font_size = 10, full_width = F)
```
	

#### Part 2: Creating a Collector's curve

Loading in my sample data.
```{r}
my_sample = data.frame(
 x=c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151),
  y=c(1,1,1,1,1,1,1,1,1,2,2,2,2,2,2,2,3,3,3,3,3,3,3,3,3,3,3,3,4,4,4,5,5,5,5,5,5,5,5,5,5,5,6,6,6,6,6,7,7,7,7,7,7,7,7,8,8,8,8,8,8,8,9,9,9,9,9,9,9,9,9,9,10,10,10,10,10,10,10,11,11,11,11,11,11,11,11,12,13,14,14,14,14,15,15,15,16,16,16,16,16,16,17,17,17,17,17,17,17,17,17,17,18,18,18,18,18,19,19,19,19,20,20,20,20,21,22,23,24,25,26,27,28,29,29,29,29,29,29,30,30,30,30,31,31,32,32,32,32,33,33)

)
```

Creating the plot.
```{r}
ggplot(my_sample, aes(x=x, y=y)) +
  geom_point() +
  geom_smooth() +
  labs(x="Cumulative number of individuals classified", y="Cumulative number of species observed")
```

* Does the curve flatten out? If so, after how many individual cells have been collected?

The curve produced continues to increase sharply as more cells are classified.

* What can you conclude from the shape of your collector's curve as to your depth of sampling?

The cap on number of species observed is limited by the sampling size: only about 2/3 of all species have been observed from this small sample

#### Part 3: Diversity estimates (alpha diversity)

##### Diversity: Simpson Reciprocal Index

$\frac{1}{D}$ where $D = \sum p_i^2$

$p_i$ = the fractional abundance of the $i^{th}$ species

For example, using the example data 1 with 3 species with 2, 4, and 1 individuals each, D =

```{r}
species1 = 29/(806)
species2 = 27/(806)
species3 = 64/(806)
species4=35/(806)
species5=59/(806)
species6=28/(806)
species7=31/(806)
species8=43/(806)
species9=42/(806)
species10=46/(806)
species11=35/(806)
species12=5/(806)
species13=5/(806)
species14=4/(806)
species15=2/(806)
species16=3/(806)
species17=14/(806)
species18=27/(806)
species19=33/(806)
species20=44/(806)
species21=38/(806)
species22=33/(806)
species23=16/(806)
species24=3/(806)
species25=1/(806)
species26=6/(806)
species27=1/(806)
species28=3/(806)
species29=1/(806)
species30=2/(806)
species31=1/(806)
species32=1/(806)
species33=2/(806)
species34=3/(806)
species35=2/(806)
species36=2/(806)
species37=1/(806)
species38=1/(806)
species39=1/(806)
species40=2/(806)
species41=3/(806)
species42=1/(806)
species43=2/(806)
species44=16/(806)
species45=16/(806)
species46=16/(806)
species47=16/(806)
species48=18/(806)
species49=19/(806)
species50=1/(806)
species51=1/(806)
species52=1/(806)

1 / (species1^2 + species2^2 + species3^2 + species4^2 + species5^2 + species6^2 + species7^2 + species8^2 + species9^2 + species10^2 + species11^2 + species12^2 + species13^2 + species14^2 + species15^2 + species16^2 + species17^2 + species18^2 + species19^2 + species20^2 + species21^2 + species22^2 + species23^2 + species24^2 + species25^2 + species26^2 + species27^2 + species28^2 + species29^2 + species30^2 + species31^2 + species32^2 + species33^2 + species34^2 + species35^2 + species36^2 + species37^2 + species38^2 + species39^2 + species40^2 + species41^2 + species42^2 + species43^2 + species44^2 + species45^2 + species46^2 + species47^2 + species48^2 + species49^2 + species50^2 + species51^2 + species52^2)

s1= 9/(151)
s2= 7/(151)
s3= 12/(151)
s4= 3/(151)
s5= 11/(151)
s6= 5/(151)
s7= 7/(151)
s8= 10/(151)
s9= 7/(151)
s10 =8/(151)
s11 =1/(151)
s12 =1/(151)
s13= 4/(151)
s14= 3/(151)
s15= 6/(151)
s16= 10/(151)
s17= 5/(151)
s18= 4/(151)
s19= 4/(151)
s20= 1/(151)
s21= 1/(151)
s22= 1/(151)
s23= 1/(151)
s24= 1/(151)
s25= 1/(151)
s26= 1/(151)
s27= 1/(151)
s28= 8/(151)
s29= 6/(151)
s30= 4/(151)
s31= 2/(151)
s32= 4/(151)
s33= 2/(151)
  
1/((s1^2) + (s2^2) +(s3^2) + (s4^2) +(s5^2) + (s6^2) +(s7^2) + (s8^2) +(s9^2) + (s10^2) + (s11^2) + (s12^2) +(s13^2) + (s14^2) +(s15^2) + (s16^2) +(s17^2) + (s18^2) +(s19^2) + (s20^2) + (s21^2) + (s22^2) +(s23^2) + (s24^2) +(s25^2) + (s26^2) +(s27^2) + (s28^2) +(s29^2) + (s30^2)+ (s31^2) + (s32^2) +(s33^2))



```


* What is the Simpson Reciprocal Index for your sample? 21.53069
* What is the Simpson Reciprocal Index for your original total community? 23.44748

##### Richness: Chao1 richness estimator

$S_{chao1} = S_{obs} + \frac{a^2}{2b})$

$S_{obs}$ = total number of species observed
a = species observed once
b = species observed twice or more


```{r}
#my community: 
52 + 12^2/(2*40)

#my sample: 
33 + 10^2/(2*23)
```

* What is the chao1 estimate for your sample? 35.17391
* What is the chao1 estimate for your original total community? 53.8

#### Part 4: Alpha-diversity functions in R

Removing the unnecesary data columns and transposing the data so that `vegan` reads it as a species table with species as columns and rows as samples.
```{r}
candy_data1_diversity = 
  candy_data %>% 
  select(name, occurences) %>% 
  spread(name, occurences)

candy_data1_diversity
```

Calculating the Simpson Reciprocal Index using the `diversity` function.
```{r}
diversity(candy_data1_diversity, index="invsimpson")
#23.44748
```

Calculating the Chao1 richness estimator (and others by default) with the the `specpool` function for extrapolated species richness.
```{r}
specpool(candy_data1_diversity)
#    Species chao chao.se jack1 jack1.se jack2 boot boot.se n
#All      52   52       0    52        0    52   52       0 1
```

repeating the calculations for my sample:
```{r}
my_sample_data = data.frame(
  number = c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33),
  name = c("M&Ms_Brown","M&Ms_Red", "M&Ms_Blue", "M&Ms_Yellow", "M&Ms_Orange", "MM&Ms_Green","Skittles_Yellow"," “Skittles_Purple", "Skittles_Orange", "Skittles_Green", "Skittles_Red", "Big_circle_Green", "Big_circle_Yellow","Red_Strings", "Jolly_Ranchers_Yellow", "Jolly_Ranchers_Green", "Jolly_Ranchers_Orange", "Jolly_Ranchers_Pink", "Jolly_Ranchers_Red", "Hershies", "Legos_BlueSquares", "Legos_PinkRectangles", "Legos_PinkSquares", "Legos_GreenRectangles", "Wine_gummies_Orange", "Wine_gummies_Yellow", "Octopus_Purple/Yellow", "Swirl_Red", "Classic_Bear_Red", "Classic_Bear_Orange", "Classic_Bear_White", "Classic_Bear_Yellow", "Classic_Bear_Green"),
  occurences = c(9,7,12,3,11,5,8,7,10,7,8,1,1,4,3,6,10,5,4,4,1,1,1,1,1,1,1,1,6,4,2,4,2)
)
my_sample_data %>% 
  kable("html") %>%
  kable_styling(bootstrap_options = "striped", font_size = 10, full_width = F)

my_sample_data1_diversity = 
  my_sample_data %>% 
  select(name, occurences) %>% 
  spread(name, occurences)

my_sample_data1_diversity

diversity(my_sample_data1_diversity, index="invsimpson")
#21.53069

specpool(my_sample_data1_diversity)
#    Species chao chao.se jack1 jack1.se jack2 boot boot.se n
#All      33   33       0    33        0    33   33       0 1

```


In Project 1, you will also see functions for calculating alpha-diversity in the `phyloseq` package since we will be working with data in that form.

For your sample:

* What are the Simpson Reciprocal Indices for your sample and community using the R function?
  community: 23.44748
  my Sample: 21.53069
* What are the chao1 estimates for your sample and community using the R function?
  community:52
  my sample:33
    + Verify that these values match your previous calculations: the Simpson Reciprocal Indices are exactly the same, but and the chao1 estimates are relatiely close.

#### Part 5: Concluding activity

* How does the measure of diversity depend on the definition of species in your samples?


* Can you think of alternative ways to cluster or bin your data that might change the observed number of species?


* How might different sequencing technologies influence observed diversity in a sample?